bin/full-build-and-upload.js

```diff
@@ -125,132 +105,72 @@ function build(callback) {
   print(`tsc${cmdSuffix}`, [], callback);
 }
 
-
-
-function injectScript(configuration, script) {
-  Object.keys(configuration).forEach((propertyName) => {
-    if (Array.isArray(script)) {
-      script = script.map(function (x) { return x.replace(new RegExp(`\\$\{${propertyName}\}`, 'g'), configuration[propertyName]); });
-    }
-    else {
-      script = script.replace(new RegExp(`\\$\{${propertyName}\}`, 'g'), configuration[propertyName]);
-    }
+function auth(clientId, clientSecret, username, password) {
+  return createRestApiClient({
+    clientId: clientId,
+    clientSecret: clientSecret,
+    username: username,
+    password: password
+  }).then(client => {
+    return client;
   });
-
-  return script;
 }
 
+function fullUpload(restApiClient, username, privateLabelId, extensions, deployFiOnly, callback) {
+  logger.multiLog(`Full uploading...`, logger.multiLogLevels.infoIntData);
 
-function inject(configuration, callback) {
-  const file = getExtensionFile();
-  const extensions = JSON.parse(file);
+  intCleanSvc.setUpService(restApiClient);
+  fieldIntDeploySvc.setUpService(restApiClient);
+  coIntDeploySvc.setUpService(restApiClient);
+  pageIntDeploySvc.setUpService(restApiClient);
 
-  if (extensions.fieldInteractions) {
-    Object.keys(extensions.fieldInteractions).forEach((entity) => {
-      if (extensions.fieldInteractions[entity]) {
-        for (let index = 0; index < extensions.fieldInteractions[entity].length; index++) {
-          extensions.fieldInteractions[entity][index].script = injectScript(configuration, extensions.fieldInteractions[entity][index].script);
+  return intCleanSvc.runEnvCleanRoutine(username, deployFiOnly).then(delResults => {
+    return fieldIntDeploySvc.deployAllFieldInteractions(privateLabelId, extensions).then(fiResults => {
+      return coIntDeploySvc.deployAllCustomObjectFieldInteractions(extensions, deployFiOnly).then(coFIResults => {
+        return pageIntDeploySvc.deployAllPageInteractions(extensions, deployFiOnly).then(piResults => ({
```

Comment:
Why do we do this callback chaining? Why not just introduce asynchronous calls and not nest everything like this?